File: gisa_controlli_ufficiali/gisa_nt/WebContent/javascript/noscia/detail.js
/*
Copyright (C) AGPL-3.0  

This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.*/
function setValore(key, value){
	var campoOld = $('#'+key);
	
	if (campoOld==null)
		return false;
	
	campoOld.replaceWith(value);
}

function setLinea(key, value){
	
	var newRow = $("<tr>");
    var cols = "";
    cols += '<td class="formLabel">LINEA</td>';
    cols += '<td>'+value+'</td>';
    newRow.append(cols);
    newRow.insertAfter($('#trLinee'));
}

function VisualizzaDettaglio(arrAnagrafica, arrLinee){

	$("div#dettaglio :input").each(function(){
		 var input = $(this); // This is the jquery object of the input, do what you will
		 if (input.attr('type') == 'button'){
			 input.hide();
		 }
		 if (input.attr('type') == 'text'){
			 input.hide();
		 }
		});
	$("#dati_impresa_id").hide();
	
	var value;
	for (var key in arrAnagrafica) {
	    value = arrAnagrafica[key];
	   setValore(key, value);
	}
	
	for (var key in arrLinee) {
	    value = arrLinee[key];
    	setLinea(key, value);
	}
	
}

File: vam/vam/src/it/us/web/action/vam/cc/decessi/Detail.java
/*
Copyright (C) AGPL-3.0  

This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.*/
package it.us.web.action.vam.cc.decessi;

import java.sql.Connection;

import javax.naming.Context;
import javax.naming.InitialContext;

import it.us.web.action.GenericAction;
import it.us.web.bean.BGuiView;
import it.us.web.bean.ServicesStatus;
import it.us.web.bean.remoteBean.RegistrazioniCaninaResponse;
import it.us.web.bean.remoteBean.RegistrazioniFelinaResponse;
import it.us.web.bean.remoteBean.RegistrazioniSinantropiResponse;
import it.us.web.bean.vam.CartellaClinica;
import it.us.web.dao.GuiViewDAO;
import it.us.web.exceptions.AuthorizationException;
import it.us.web.util.sinantropi.SinantropoUtil;
import it.us.web.util.vam.CaninaRemoteUtil;
import it.us.web.util.vam.FelinaRemoteUtil;



public class Detail extends GenericAction {

	
	public void can() throws AuthorizationException
	{
		BGuiView gui = GuiViewDAO.getView( "CC", "DETAIL", "MAIN" );
		can( gui, "w" );
	}
	
	@Override
	public void setSegnalibroDocumentazione()
	{
		setSegnalibroDocumentazione("decesso");
	}
	
	@Override
	public void canClinicaCessata() throws AuthorizationException
	{
	}

	public void execute() throws Exception
	{
		Context ctx = new InitialContext();
		javax.sql.DataSource ds = (javax.sql.DataSource)ctx.lookup("java:comp/env/jdbc/bduS");
		Connection connection = ds.getConnection();
		GenericAction.aggiornaConnessioneApertaSessione(req);
		setConnectionBdu(connection);	
		/*SERVIZIO DECESSO*/
		ServicesStatus status = new ServicesStatus();
		
		if (cc.getAccettazione().getAnimale().getLookupSpecie().getId() == 1 && !cc.getAccettazione().getAnimale().getDecedutoNonAnagrafe()) {
			RegistrazioniCaninaResponse res	= CaninaRemoteUtil.getInfoDecesso( cc.getAccettazione().getAnimale().getIdentificativo(), utente, status, connection ,req);
			req.setAttribute("res", res);
			
			//Errore nella comunicazione con il Wrapper
			if (!status.isAllRight()) {
				setErrore("Errore nella comunicazione con la BDR di riferimento");
				goToAction(new it.us.web.action.vam.cc.Detail());
			}
			if (res == null) {
				goToAction(new ToAdd());
			}
		
		}
		else if (cc.getAccettazione().getAnimale().getLookupSpecie().getId() == 2 && !cc.getAccettazione().getAnimale().getDecedutoNonAnagrafe()) {
			RegistrazioniFelinaResponse rfr = FelinaRemoteUtil.getInfoDecesso( cc.getAccettazione().getAnimale().getIdentificativo(), utente, status, connection ,req);
			req.setAttribute("res", rfr);
			
			//Errore nella comunicazione con il Wrapper
			if (!status.isAllRight()) {
				setErrore("Errore nella comunicazione con la BDR di riferimento");
				goToAction(new it.us.web.action.vam.cc.Detail());
			}
			if (rfr == null) {
				goToAction(new ToAdd());
			}
		}
		else if (cc.getAccettazione().getAnimale().getLookupSpecie().getId() == 3 && !cc.getAccettazione().getAnimale().getDecedutoNonAnagrafe()) {
			RegistrazioniSinantropiResponse rsr = SinantropoUtil.getInfoDecesso(persistence, cc.getAccettazione().getAnimale());
			req.setAttribute("res", rsr);
			
			if (rsr == null) {
				goToAction(new ToAdd());
			}
		}
//		else if (cc.getAccettazione().getAnimale().getDecedutoNonAnagrafe() && cc.getAccettazione().getAnimale().getDataMorte() == null) {
//			goToAction(new ToAdd());
//		}
		
		gotoPage("/jsp/vam/cc/decessi/detail.jsp");
				
	}
	
	
}
